File: exercism/c/grains/grains.c
#include "grains.h"
#include <stdint.h>

uint64_t square(uint8_t index) {
  if (index < 1 || index > 64)
    return 0;
  return (uint64_t)1 << (index - 1);
}

uint64_t total(void) { return ~0; }

File: exercism/zig/grains/grains.zig
const std = @import("std");

pub const ChessboardError = error{IndexOutOfBounds};

pub fn square(index: usize) ChessboardError!u64 {
    if (index == 0 or index > 64) return ChessboardError.IndexOutOfBounds;
    return @as(usize, 1) << @intCast(index - 1);
}

pub fn total() u64 {
    return std.math.maxInt(u64);
}
